### PR TITLE
ambient: don't annotate pods on delete

### PR DESCRIPTION
**Pull request by dddddai**
**Please provide a description of this PR:**
Each time a pod is deleted, istio-cni reports an error:
```
failed to annotate pod unenrollment: pods "xxx" not found
```
We should not annotate the pod if it no longer exists

### CODE REVIEW

File: cni/pkg/ambient/net.go [hunk 1]
```diff
@@ -287,6 +285,8 @@ func (s *Server) DelPodFromMesh(pod *corev1.Pod) {
 		if err := s.delPodEbpfOnNode(pod.Status.PodIP); err != nil {
 			log.Errorf("failed to del POD ebpf: %v", err)
 		}
+	}
+	if event.New != nil {
```
[the remark this answers is not in the document]
Comment (dddddai) [reply cni/pkg/ambient/net.go:290]: > Under what scenario would `New` be nil? Can that be mentioned in a quick comment?

Added a comment

> Given that `AnnotateUnenrollPod` now ignores un-annotatable pods, do we really need to pass the whole `event` to this function just to short-circuit that eventuality?

It just ignores pods without ambient redirection enabled. 
When a pod in ambient is deleted, it still sends a request to the api server trying to annotate the deleted pod, then gets a "not found" error
https://github.com/istio/istio/blob/40c90f2ae557028045640d8f36e4a5b2f5b3b612/cni/pkg/ambient/net.go#L134-L137

File: cni/pkg/ambient/informers.go [hunk 2]
```diff
@@ -62,27 +62,20 @@ func (s *Server) EnqueueNamespace(o controllers.Object) {
 	matchAmbient := o.GetLabels()[constants.DataplaneMode] == constants.DataplaneModeAmbient
 	if matchAmbient {
 		log.Infof("Namespace %s is enabled in ambient mesh", namespace)
-		for _, pod := range s.pods.List(namespace, klabels.Everything()) {
+	} else {
+		log.Infof("Namespace %s is disabled from ambient mesh", namespace)
+	}
+	for _, pod := range s.pods.List(namespace, klabels.Everything()) {
```
Comment (hzxuzhonghu) [inline cni/pkg/ambient/informers.go:68]: I think we may meet race here in case namespace deleting. For this case, the pod should be `Delete`

Comment (dddddai) [reply cni/pkg/ambient/informers.go:77]: Yeah we don't need to handle ns delete events

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by hzxuzhonghu** [comment]
does it need to cherrypick to 1.18?

**Pull request merged**